examples/clean-architecture/controllers/note.controller.js

```diff
@@ -0,0 +1,57 @@
+var path = require("path");
+var Note = require(path.join(__dirname, "..", "entities", "note.entity"));
+
+/**
+ * Please, note that here I'm not using ES6 classes just for compatibility purposes
+ * In a real-world application, you should use ES6 classes
+ */
+function NoteController(noteUseCase) {
+  this.noteUseCase = noteUseCase;
+}
+
+NoteController.prototype.getAll = function (req, res) {
+  var notes = this.noteUseCase.getAll();
+  res.json(notes);
+};
+
+NoteController.prototype.getById = function (req, res) {
+  try {
+    var noteId = parseInt(req.params.id);
+    var note = this.noteUseCase.getById(noteId);
+    res.json(note);
+  } catch (error) {
+    res.status(404).json(error.message);
```

Comment:
I'm not sure if the examples should demonstrate that any possible error above should result in a 400. What if the data source to get the note has an issue (like cannot connect to db)? One would normally expect it to 500 rather than 400. I also wonder if these should be using the Express error handling mechanism to respond rather than demo putting a try...catch in every single handler?

Reply:
In a production environment we will of course differenciate between 400 and 500 errorrs. Since this is a in-memory datastore just to show the pattern the db connection cannot fail and I don't want to add more boilerplate to "simulate" db connection failures (since this is not the goal of the example). 

I had used the try/catch also in the getAll but since the getAll function is sync and is not a real call to db cannot fail (at most returns an empty array) so, in order to mantain the code coverage, I have removed the try/catch block from here.

"ideally we want to demo using Express.js and it's features in our demo apps as much as possible" are you advicing to use the default next(err) error handler? If so, I think it is a great idea! I'll update the code accordingly :)